squidpy/read/_utils.py

```diff
@@ -0,0 +1,83 @@
+from __future__ import annotations
+
+from h5py import File
+from types import MappingProxyType
+from typing import Any, Mapping, Optional, Sequence
+from imageio import imread
+from pathlib import Path
+
+from scanpy import read_10x_h5
+from anndata import AnnData, read_mtx, read_text
+
+import pandas as pd
+
+from squidpy._utils import NDArrayA
+from squidpy._constants._pkg_constants import Key
+
+
+def _read_count(
+    path: str | Path,
+    count_file: str,
+    genome: Optional[str] = None,
+    library_id: Optional[str] = None,
+    h5_kwargs: Mapping[str, Any] = MappingProxyType({}),
+    text_kwargs: Mapping[str, Any] = MappingProxyType({}),
+    mtx_kwargs: Mapping[str, Any] = MappingProxyType({}),
+) -> AnnData:
+    """Load count files."""
+    path = Path(path)
+    if count_file.endswith(".h5"):
+        adata = read_10x_h5(path / count_file, genome=genome, **h5_kwargs)
+        with File(path / count_file, mode="r") as f:
+            attrs = dict(f.attrs)
+            if library_id is None:
+                try:
+                    library_id = str(attrs.pop("library_ids")[0], "utf-8")
+                except ValueError:
+                    raise ValueError(f"Invalid value for `library_id: {library_id}`. Cannot be None.")
+            adata.uns[Key.uns.spatial] = {library_id: {}}
+            metadata_dic = {
+                k: (str(attrs[k], "utf-8") if isinstance(attrs[k], bytes) else attrs[k])
+                for k in ("chemistry_description", "software_version")
+                if k in attrs
+            }
+            adata.uns[Key.uns.spatial][library_id]["metadata"] = metadata_dic
```

Comment:
Is this info always going to exist?

I think I would be fine with `read_10x_h5` also reading this (though perhaps not under the spatial key).